src/VisualStudio/Core/Impl/Options/Style/NamingPreferences/NamingRuleDialog.xaml.cs

```diff
@@ -0,0 +1,119 @@
+        }
+
+        private void CreateSymbolSpecification(object sender, RoutedEventArgs e)
+        {
+            var newSymbolSpecificationViewModel = new SymbolSpecificationViewModel(_namingStylesViewModel.LanguageName, _notificationService);
+            var dialog = new SymbolSpecificationDialog(newSymbolSpecificationViewModel);
+            var result = dialog.ShowModal();
+            if (result == true)
+            {
+                _namingStylesViewModel.AddSymbolSpecification(newSymbolSpecificationViewModel);
+                _viewModel.SelectedSymbolSpecificationIndex = _viewModel.SymbolSpecificationList.IndexOf(newSymbolSpecificationViewModel);
+            }
+        }
+
+        private void ConfigureSymbolSpecifications(object sender, RoutedEventArgs e)
+        {
+            if (_viewModel.SelectedSymbolSpecificationIndex >= 0)
+            {
+                var symbolSpecificationViewModel = _viewModel.SymbolSpecificationList.GetItemAt(_viewModel.SelectedSymbolSpecificationIndex) as SymbolSpecificationViewModel;
+                var symbolSpecificationClone = new SymbolSpecificationViewModel(_namingStylesViewModel.LanguageName, symbolSpecificationViewModel.GetSymbolSpecification(), _notificationService);
+
+                var dialog = new SymbolSpecificationDialog(symbolSpecificationClone);
+                var result = dialog.ShowModal();
+                if (result == true)
+                {
+                    symbolSpecificationViewModel.ModifierList = symbolSpecificationClone.ModifierList;
+                    symbolSpecificationViewModel.SymbolKindList = symbolSpecificationClone.SymbolKindList;
+                    symbolSpecificationViewModel.AccessibilityList = symbolSpecificationClone.AccessibilityList;
+                    symbolSpecificationViewModel.SymbolSpecName = symbolSpecificationClone.SymbolSpecName;
+                }
+            }
+        }
+
+        private void CreateNamingStyle(object sender, RoutedEventArgs e)
+        {
+            var newNamingStyleViewModel = new NamingStyleViewModel(new NamingStyle { ID = Guid.NewGuid() }, _notificationService);
+            var dialog = new NamingStyleDialog(newNamingStyleViewModel);
+            var result = dialog.ShowModal();
+            if (result == true)
+            {
+                _namingStylesViewModel.AddNamingStyle(newNamingStyleViewModel);
+                _viewModel.NamingStyleIndex = _viewModel.NamingStyleList.IndexOf(newNamingStyleViewModel);
+            }
+        }
+
+        private void ConfigureNamingStyles(object sender, RoutedEventArgs e)
+        {
+            if (_viewModel.NamingStyleIndex >= 0)
+            {
+                var namingStyleMutable = _viewModel.NamingStyleList.GetItemAt(_viewModel.NamingStyleIndex) as NamingStyleViewModel;
+
+                var style = namingStyleMutable.GetNamingStyle();
+                var styleClone = new NamingStyle
+                {
+                    ID = style.ID,
```

Comment:
not a huge fan of object initializers as they mean hte objects are mutable.  Do we have to have that?